scipy/stats/_continuous_distns.py

```diff
@@ -6056,11 +6057,23 @@ def _logpdf(self, x, skew):
         return ans
 
     def _cdf(self, x, skew):
-        ans, x, transx, mask, invmask, _, alpha, _ = (
+        ans, x, transx, mask, invmask, beta, alpha, zeta = (
             self._preprocess(x, skew))
 
         ans[mask] = _norm_cdf(x[mask])
-        ans[invmask] = gamma._cdf(transx, alpha)
+        invmask1 = np.logical_and(invmask, skew > 0)
+        transx1 = beta[invmask1] * (x[invmask1] - zeta[invmask1])
+        ans[invmask1] = gamma._cdf(transx1, alpha[invmask1])
+
+        # The gamma._cdf approach wasn't working with negative skew.
+        # Note that multiplying the skew by -1 reflects about x=0.
+        # So instead of evaluating the CDF with negative skew at x,
+        # evaluate the SF with positive skew at -x.
+        invmask2 = np.logical_and(invmask, skew < 0)
+        transx2 = -beta[invmask2] * (-x[invmask2] + zeta[invmask2])
+        # gamma._sf produces NaNs when transx2 < 0, so use gamma.sf
+        ans[invmask2] = gamma.sf(transx2, alpha[invmask2])
```

Comment:
I'm not sure what was going on with gamma.cdf when the skew was negative. That absolute value sign in the definition of the pearson3 PDF probably throws a wrench into things. Anyway, this makes sense and works, but if you can think of a cleaner way to do this, I'm happy to change it.